store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaOverlapTest.java

```diff
@@ -170,17 +181,12 @@ public void shouldOverwriteTypeDefinitionWhenConflictingSchemasMerge() throws Op
                 .build();
         addGraphWith(GRAPH_ID_B, graphBSchema);
 
-        // When
-        final Schema schema = federatedStore.execute(new GetSchema.Builder().build(), testContext);
-
-        // Then
-        assertThat(schema.validate().isValid())
-                .withFailMessage(schema.validate().getErrorString())
-                .isTrue();
-        assertThat(schema.getType(INTEGER)).isEqualTo(STRING_TYPE);
-        assertThat(schema.toString())
-                .isNotEqualTo(GRAPH_A_SCHEMA.toString())
-                .isEqualTo(graphBSchema.toString());
+        // When then
+        assertThatException()
+                .isThrownBy(() -> federatedStore.execute(new GetSchema.Builder().build(), testContext))
```

Comment:
With these, it might be an idea to include a check on the type of exception thrown. To ensure these tests are revisited if how the exception is thrown gets changed.

Reply:
I was going to, but then I didn't think it matters if it gets caught and re-thrown. The value is in the stack trace messages. You think is valuable? I did consider it.

Reply:
It depends. I was thinking if someone edits the exception type then it would probably be a good idea to revisit the tests at the same time. Checking the exception type will ensure that happens. This might not be something we want to do as it's not strictly important what the type is.

Reply:
It throws SchemaExceptions and by the time the test catches it I think it becomes an Operation Exception. Due to the many rethrows and adding more info to exceptions.